packages/core/src/external/aws/lambda-logic/bundle-to-brief.ts

```diff
@@ -1,128 +1,746 @@
+
+Don't tell me that you are writing a summary, just write the summary. Also, don't tell me about any limitations of the information provided.
+
+SUMMARY:
+`;
+  const SUMMARY_PROMPT_REFINED = PromptTemplate.fromTemplate(summaryTemplateRefined);
+  const summaryChainRefined = new StuffDocumentsChain({
+    llmChain: new LLMChain({
+      llm: llmSummary as any, // eslint-disable-line @typescript-eslint/no-explicit-any
+      prompt: SUMMARY_PROMPT_REFINED as any, // eslint-disable-line @typescript-eslint/no-explicit-any
+    }),
+    documentVariableName: "text",
+  });
+
+  const mapReduce = new MapReduceDocumentsChain({
+    llmChain: summaryChain,
+    combineDocumentChain: summaryChainRefined,
+    documentVariableName: "text",
+    verbose: true,
+  });
+
+  const summary = (await mapReduce.invoke({
+    input_documents: docs,
+  })) as { text: string };
+
+  if (!summary.text) return undefined;
+  return summary.text;
 }
 
-export async function bundleToBrief(
-  fhirBundle: Bundle,
-  cxId: string,
-  patientId: string
-): Promise<string | undefined> {
-  const { log } = out(`AI Brief for cx ${cxId}, patient ${patientId}`);
-  let processedBundleInput = await prepareBundleForBrief(fhirBundle);
-  if (!processedBundleInput) return undefined;
-
-  if (processedBundleInput.length > MAXIMUM_BRIEF_STRING_LENGTH) {
-    const initialLength = processedBundleInput.length;
-    processedBundleInput = processedBundleInput.slice(0, MAXIMUM_BRIEF_STRING_LENGTH);
-    const msg = `Brief string input was truncated`;
-    log(`${msg}. Initial length was: ${initialLength}`);
-    capture.message(msg, {
-      extra: {
-        patientId,
-        cxId,
-        stringLength: processedBundleInput.length,
-        maximumLength: MAXIMUM_BRIEF_STRING_LENGTH,
-      },
-      level: "info",
+export function prepareBundleForBrief(bundle: Bundle): string | undefined {
+  if (bundle.entry?.length === 0) return undefined;
+  const filteredString = filterBundleResources(bundle);
+  return filteredString;
+}
+
+function filterBundleResources(bundle: Bundle) {
+  const patient = findPatientResource(bundle);
+  if (!patient) return undefined;
+
+  // First pass to remove a ton of useless stuff and apply resource-specific modifications
+  const leanBundleEntries = buildSlimmerBundle(bundle);
+
+  // Build a map of these lean resources for cross-referencing
+  const resourceMap = new Map<string, Resource>();
+  leanBundleEntries?.forEach(res => {
+    if (!res || !res.id) return;
+    const mapKey = `${res.resourceType}/${res.id}`;
+    resourceMap.set(mapKey, res);
+  });
+
+  // Replace the references with actual data and collect references for embedded resources
+  const containedResourceIdsSet = new Set<string>();
+  const processedEntries = leanBundleEntries?.map(res => {
+    const { updRes, ids } = replaceReferencesWithData(res, resourceMap);
+    ids.forEach(id => containedResourceIdsSet.add(id));
```

Comment:
could you do the uuid pop on 190 here? and simply have the  `containedResourceIds` without the second iteration?

Reply:
Good point, I'll add as a TODO on the downstream PR

Reply:
done